src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/GerritProject.java

```diff
@@ -219,29 +220,22 @@ public boolean isInteresting(String project, String branch, String topic, List<S
                 boolean foundInterestingTopicOrFile = false;
                 if (b.isInteresting(branch)) {
                     if (forbiddenFilePaths != null) {
-                        for (FilePath ffp : forbiddenFilePaths) {
-                            if (ffp.isInteresting(files)) {
-                                foundInterestingForbidden = true;
-                                break;
+                        Iterator<String> i = files.iterator();
+                        while (i.hasNext()) {
+                            String file = i.next();
+                            for (FilePath ffp : forbiddenFilePaths) {
+                                if (ffp.isInteresting(file)) {
+                                    if (!disableStrictForbiddenFileVerification) {
+                                        return false;
+                                    } else {
+                                        i.remove();
+                                        break;
+                                    }
+                                }
                             }
                         }
                     }
-                    if (isInterestingTopic(topic) && isInterestingFile(files)) {
-                        foundInterestingTopicOrFile = true;
-                    }
-                    if (disableStrictForbiddenFileVerification) {
-                        // Here we want to be able to trigger a build if the event contains
-                        // wanted topics or file paths even though there may be a forbidden file
-                        return foundInterestingTopicOrFile;
-                    } else {
-                        if (foundInterestingForbidden) {
-                            // we have a forbidden file and a wanted file path.
-                            return false;
-                        } else if (foundInterestingTopicOrFile) {
-                            // we DO not have a forbidden file and but we have a wanted file path.
-                            return true;
-                        }
-                    }
+                    return isInterestingTopic(topic) && !files.isEmpty() && isInterestingFile(files);
```

Comment:
I'm not sure we want to return `false` if there are no changed files. Only if there are files we are interested in and changed files is empty.

Reply:
The condition you indicate is handled in isInterestingFile().

This test is here because we should return `false` when all files are forbidden, i.e. there are no more after filtering.

Reply:
Done